src/common/options/global.yaml.in

```diff
@@ -3682,6 +3682,29 @@ options:
   min: 896_M
   flags:
   - runtime
+- name: osd_memory_thread_cache_bytes
+  type: size
+  level: basic
+  desc: When tcmalloc is enabled it specifies thread cache size
+  long_desc: Sets tcmalloc.max_total_thread_cache_bytes value.
+    If set this will override settings privided via environment variable
+    TCMALLOC_MAX_TOTAL_THREAD_CACHE_BYTES.
+  fmt_desc: TCMalloc assigns each thread a thread-local cache.
+    Small allocations are satisfied from the thread-local cache.
+    Objects are moved from central data structures into a thread-local cache as needed,
+    and periodic garbage collections are used to migrate memory back from
+    a thread-local cache into the central data structures.
+    Programs with intensive use of dynamic memory for thread-local, short-lived objects
+    do benefit greately from having large cache.
+    TCMalloc default - 32_M. TCMalloc minimum - 512_K. Recommended for OSD - 128_M.
```

Comment:
It's looking like 128_M is also a very reasonable default for rgw as well (not sure about the mon or mds, but I doubt it will hurt).  I'm thinking 128_M is probably a good global default for us.

Reply:
I personally do not have opinion whether this should be a family of settings or one global setting.
@markhpc @neha-ojha @mattbenjamin please advise.

Reply:
Given that we have already merged https://github.com/ceph/ceph/pull/41805, which should ideally take care of things globally, I see this PR as an extra level of caution in the OSD to appropriately set things. We can always extend this logic incrementally across daemons.

Reply:
@neha-ojha #41805 is probably the wrong way to solve this in the long run imho.  This isn't the first time that these settings haven't been set correctly higher up in the stack (And I suspect that it won't be the last).  The advantage of Adam's approach (assuming that we can use it for all daemons) is that it follows the same approach as the rest of our settings.  It's very clear what the default is and so long as we document how environment variable overrides work (or don't work) I think we'll see fewer bugs long term vs relying on ceph-adm/ceph-ansible/ceph-deploy/cbt/mkcephfs/* to do the right thing.